### ISSUE TITLE
有关DAAN参数gamma、两个A-distance和w估计值计算的问题

### ISSUE THREAD

**Issue by Antinomy20001** [jindongwang/transferlearning#172]
您好！您的代码对我的学习起到了很大的帮助，十分感谢！最近我在学习这个repo关于DAAN的实现时对参数gamma有一些疑惑：

https://github.com/jindongwang/transferlearning/blob/c37ab0d0b9fa17ef8c51097acd8420ebbe81750b/code/deep/DAAN/train.py#L44-L45

https://github.com/jindongwang/transferlearning/blob/c37ab0d0b9fa17ef8c51097acd8420ebbe81750b/code/deep/DAAN/train.py#L146-L150

从paper来看，DAAN的Loss应该是：

$$L = L_y - \lambda \left( \left( 1 - \omega \right) L_g + \omega L_l \right )$$

从argparse list来看只有这个λ是没有出现的，并且arg.gamma并没有在哪里的计算被使用过，同时`loss = soft_loss + join_loss`也没有λ计算进去，因此请问这个gamma是否就是指公式中的λ吗，如果是这样的话，argparse中对于gamma的描述似乎也是复制粘贴diff_lr的，那么这应该是个bug吧。

**Comment by Antinomy20001** [comment]
另外，关于paper中提到的参数Omega的计算应该是基于代码中`D_C`和`D_M`来计算，并且Omega应该在0和1之间。

但是在A-distance的公式：$D = 2(1-2*err)$，因为暂时没有找到说A-Distance可不可以是负值的文献，我猜想应该是不能为负值的，但在代码里如果err也就是local domain dicriminator的loss如果大于0.5那就会变成负值了，也会导致计算出来的Omega，也就是代码里的`MU`大于1。但我将每个epoch的`D_C`和`D_M`取绝对值的话，最后整个代码训练的结果来看是work的。

**Comment by jindongwang** [comment]
@Richardych

**Comment by Richardych** [comment]
@Antinomy20001 
gamma只是前期实验的时候用的，本质就是lambda，lambda=1，可以不引入。
A-distance按公式有可能为负，但我实验时发现MU处于0-1之间，你可以根据你的实验加一些约束，比如对err做一下clip等

**Comment by Antinomy20001** [comment]
> @Antinomy20001
> gamma只是前期实验的时候用的，本质就是lambda，lambda=1，可以不引入。
> A-distance按公式有可能为负，但我实验时发现MU处于0-1之间，你可以根据你的实验加一些约束，比如对err做一下clip等

感谢回复！好的，可能是因为我的数据集需要对discriminator的结构调整一下，另外还有一个地方：
https://github.com/jindongwang/transferlearning/blob/c37ab0d0b9fa17ef8c51097acd8420ebbe81750b/code/deep/DAAN/train.py#L139-L140

请问这里的0.05和0.01的常数是如何得到的？

**Comment by Richardych** [comment]
> > @Antinomy20001
> > gamma只是前期实验的时候用的，本质就是lambda，lambda=1，可以不引入。
> > A-distance按公式有可能为负，但我实验时发现MU处于0-1之间，你可以根据你的实验加一些约束，比如对err做一下clip等
> 
> 感谢回复！好的，可能是因为我的数据集需要对discriminator的结构调整一下，另外还有一个地方：
> https://github.com/jindongwang/transferlearning/blob/c37ab0d0b9fa17ef8c51097acd8420ebbe81750b/code/deep/DAAN/train.py#L139-L140
> 
> 请问这里的0.05和0.01的常数是如何得到的？

为了调整loss的量级，有利于训练收敛
